adapter/memQuota/memQuota.go

```diff
@@ -296,46 +289,65 @@ func (mq *memQuota) reapDedup() {
 	}
 }
 
-// Produce a unique key representing the cell
-func makeKey(args adapter.QuotaArgs) string {
+// Produce a unique key representing the given labels.
+func makeKey(name string, labels map[string]interface{}) string {
 	ws := keyWorkspacePool.Get().(*keyWorkspace)
 	keys := ws.keys
 	buffer := ws.buffer
 
-	// get all the label names
-	for k := range args.Labels {
+	// ensure stable order
+	for k := range labels {
 		keys = append(keys, k)
 	}
-
-	// ensure stable order
 	sort.Strings(keys)
 
-	buffer = append(buffer, []byte(args.Name)...)
+	buffer.WriteString(name)
 	for _, k := range keys {
-		buffer = append(buffer, []byte(";")...)
-		buffer = append(buffer, []byte(k)...)
-		buffer = append(buffer, []byte("=")...)
+		buffer.WriteString(";")
+		buffer.WriteString(k)
+		buffer.WriteString("=")
 
-		switch v := args.Labels[k].(type) {
+		switch v := labels[k].(type) {
```

Comment:
The code is currently written to specifically avoid generating temp strings for better perf, so I'd rather not backslide.